src/services/railcontent.js

```diff
@@ -239,6 +239,36 @@ export async function fetchContentPageUserData(contentId) {
     }
 }
 
+/**
+ * Fetches the ID and Type of the piece of content that would be the next one for the user
+ *
+ * @param {int} contentId - The id of the parent (method, level, or course) piece of content.
+ * @returns {Promise<Object|null>} - Returns and Object with the id and type of the next piece of content if found, otherwise null.
+ */
+export async function fetchNextContentDataForParent(contentId) {
+    let url = `/content/${contentId}/next/${globalConfig.railcontentConfig.userId}`;
+    const headers = {
+        'Content-Type': 'application/json',
+        'X-CSRF-TOKEN': globalConfig.railcontentConfig.token
+    };
+
+    try {
+        const response = await fetchAbsolute(url, {headers});
+        const result = await response.json();
+        if (result) {
+            // console.log('fetchNextContentDataForParent', result);
+            return result.next;
+        } else {
+            console.log('fetchNextContentDataForParent result not json');
+            return null;
+        }
+    } catch (error) {
+        console.error('Fetch error:', error);
+        return null;
+    }
```

Comment:
to avoid the duplication. The above functions should also use this, but here we are.
```suggestion
    return fetchHandler(url);
```

Reply:
That is a much better idea. Sounds like a nice and hopefully pretty small refactor for next year